### ISSUE TITLE
showDatePicker occupying almost the whole screen

### ISSUE THREAD

**Issue by vinicius-ps** [flutter/flutter#46608]
## Steps to Reproduce
1. Try to use showDatePicker with a high resolution
2. Datepicker will occupy almost the whole screen

**Target Platform:** Mobile, Web
**Target OS version/browser:** Android 9, Google Chrome 78

### Code
```
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

```
```
class _MyHomePageState extends State<MyHomePage> {
  DateTime selectedDate = DateTime.now();
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: Center(
        child: FloatingActionButton(
          onPressed: () async {
                final DateTime picked = await showDatePicker(
        context: context,
        initialDate: selectedDate,
        firstDate: DateTime(2015, 8),
        lastDate: DateTime(2101));

    if (picked != null && picked != selectedDate)
      setState(() {
        selectedDate = picked;
      });
          },
          child: Icon(Icons.add),
        ),
      ),
    );
  }

}
```

## Doctor
[✓] Flutter (Channel master, v1.13.1-pre.63, on Linux, locale en_US.UTF-8)
    • Flutter version 1.13.1-pre.63 at /home/vinicius/development/flutter
    • Framework revision 5e06984854 (5 hours ago), 2019-12-09 04:54:33 -0500
    • Engine revision 4beaa696fc
    • Dart version 2.7.0 (build 2.7.0-dev.2.1 a9c77229c2)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /home/vinicius/Android/Sdk
    • Android NDK location not configured (optional; useful for native profiling
      support)
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /home/vinicius/development/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Android Studio (version 3.5)
    • Android Studio at /home/vinicius/development/android-studio
    • Flutter plugin version 39.0.3
    • Dart plugin version 191.8423
    • Java version OpenJDK Runtime Environment (build
      1.8.0_202-release-1483-b49-5587405)

[✓] Connected device (3 available)
    • Android SDK built for x86 • emulator-5554 • android-x86    • Android 9
      (API 28) (emulator)
    • Chrome                    • chrome        • web-javascript • Google Chrome
      78.0.3904.108
    • Web Server                • web-server    • web-javascript • Flutter Tools

• No issues found!

![Screenshot from 2019-12-09 11-48-15](https://user-images.githubusercontent.com/27825827/70455832-b81e6f00-1a8b-11ea-9238-fb805f1850c0.png)
![Screenshot from 2019-12-09 11-52-07](https://user-images.githubusercontent.com/27825827/70455843-bd7bb980-1a8b-11ea-9415-0f958a07b3ff.png)

**Comment by HansMuller** [comment]
CC @darrenaustin

**Comment by yjbanov** [comment]
@HansMuller Do you believe this is a web-only issue? If the fix is purely in the framework, we prefer to not put the platform-web label.

**Comment by HansMuller** [comment]
@yjbanov I don't think it's a web-only issue, it's really just a big screen issue. 

The date picker classes are being revised (see https://github.com/flutter/flutter/issues/43035). When that work gets underway again early next year, we should be able to address this issue.

**Comment by darrenaustin** [comment]
This should be fixed with #50546.
